axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/PaymentSessionValidateBankPaymentServiceImpl.java

```diff
@@ -0,0 +1,286 @@
+    bankOrderRepo.save(bankOrder);
+
+    if (paymentSession.getPaymentMode().getAutoConfirmBankOrder()) {
+      try {
+        bankOrderService.confirm(bankOrder);
+      } catch (JAXBException | IOException | DatatypeConfigurationException e) {
+        throw new AxelorException(
+            TraceBackRepository.CATEGORY_INCONSISTENCY, e.getLocalizedMessage());
+      }
+    }
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  protected BankOrder generateBankOrderFromPaymentSession(PaymentSession paymentSession)
+      throws AxelorException {
+    BankOrder bankOrder = this.createBankOrder(paymentSession);
+
+    paymentSession.setBankOrder(bankOrder);
+    bankOrderService.generateSequence(bankOrder);
+
+    return bankOrder;
+  }
+
+  protected BankOrder createBankOrder(PaymentSession paymentSession) throws AxelorException {
+    BankOrder bankOrder =
+        bankOrderCreateService.createBankOrder(
+            paymentSession.getPaymentMode(),
+            paymentSession.getPartnerTypeSelect(),
+            paymentSession.getPaymentDate(),
+            paymentSession.getCompany(),
+            paymentSession.getBankDetails(),
+            paymentSession.getCurrency(),
+            paymentSession.getSequence(),
+            this.getLabel(paymentSession),
+            BankOrderRepository.TECHNICAL_ORIGIN_AUTOMATIC);
+
+    if (!paymentSession.getCurrency().equals(paymentSession.getCompany().getCurrency())) {
+      bankOrder.setIsMultiCurrency(true);
+    }
+
+    return bankOrder;
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  protected PaymentSession processInvoiceTerm(
+      PaymentSession paymentSession,
+      InvoiceTerm invoiceTerm,
+      Map<Partner, List<Move>> moveMap,
+      Map<Move, BigDecimal> paymentAmountMap,
+      boolean out,
+      boolean isGlobal)
+      throws AxelorException {
+    paymentSession =
+        super.processInvoiceTerm(
+            paymentSession, invoiceTerm, moveMap, paymentAmountMap, out, isGlobal);
+
+    if (paymentSession.getBankOrder() != null) {
+      this.createOrUpdateBankOrderLineFromInvoiceTerm(
+          paymentSession, invoiceTerm, paymentSession.getBankOrder());
+    }
+
+    return paymentSession;
+  }
+
+  protected void createOrUpdateBankOrderLineFromInvoiceTerm(
+      PaymentSession paymentSession, InvoiceTerm invoiceTerm, BankOrder bankOrder)
+      throws AxelorException {
+    BankOrderLine bankOrderLine = null;
+
+    if (paymentSession.getPaymentMode().getConsoBankOrderLinePerPartner()) {
+      bankOrderLine =
+          bankOrder.getBankOrderLineList().stream()
+              .filter(
+                  it ->
+                      it.getPartner().equals(invoiceTerm.getMoveLine().getPartner())
+                          && ((it.getReceiverBankDetails() == null
+                                  && invoiceTerm.getBankDetails() == null)
+                              || it.getReceiverBankDetails().equals(invoiceTerm.getBankDetails())))
```

Comment:
Here I think we can still have a NPE if it.getReceiverBankDetails() is null and invoiceTerm.getBankDetails() is not null